x/chainlink/keeper/keeper.go

```diff
@@ -437,3 +437,15 @@ func (k Keeper) FeedOwnershipTransfer(ctx sdk.Context, feedOwnershipTransfer *ty
 
 	return ctx.BlockHeight(), ctx.TxBytes(), nil
 }
+
+func (k Keeper) RequestNewRound(ctx sdk.Context, requestNewRound *types.MsgRequestNewRound) (int64, []byte, error) {
+	// might be possible that this method can throw so might be best to move this to the very end.
+	roundStore := ctx.KVStore(k.roundStoreKey)
+	currentLatestRoundId := k.GetLatestRoundId(ctx, requestNewRound.FeedId)
+	roundId := currentLatestRoundId + 1
+
+	// update the latest roundId of the current feedId
+	roundStore.Set(types.GetRoundIdKey(requestNewRound.GetFeedId()), i64tob(roundId))
```

Comment:
I'm not sure if we should set the new round in store here since the feed owner only sends a tx to request, but the actual feed data has not been submitted

Reply:
there should be a check in ante to see if the feedid has been set already (if feedid == nil, should throw). but i can verify and get back to you on that

Reply:
You right, I think we should just emit an event with the next supposed round id of the feed, without storing it since no feed data is provided for this one (in the case of user request feed data of the last round id, nothing will be returned).

Reply:
```
case *types.MsgRequestNewRound:
			feed := fd.chainLinkKeeper.GetFeed(ctx, t.GetFeedId())
			if feed.Feed.Empty() {
				return ctx, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, ErrFeedDoesNotExist)
			}
			signer := t.GetSigners()[0]
			if !feed.GetFeed().GetFeedOwner().Equals(signer) {
				return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, ErrSignerIsNotFeedOwner, common.BytesToAddress(signer.Bytes()), signer)
			}
```
the ante handler should verify if the feed exists before it updates the `roundId`